staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go

```diff
@@ -741,10 +741,10 @@ func TestTransformationFailure(t *testing.T) {
 	}
 
 	// Delete succeeds but reports an error because we cannot access the body
-	if err := store.Delete(ctx, preset[1].key, &example.Pod{}, nil, storage.ValidateAllObjectFunc); !storage.IsInternalError(err) {
+	if err := store.Delete(ctx, preset[1].key, &example.Pod{}, nil, storage.ValidateAllObjectFunc); !storage.IsTransformerError(err) {
 		t.Errorf("Unexpected error: %v", err)
 	}
-	if err := store.Get(ctx, preset[1].key, "", &example.Pod{}, false); !storage.IsInternalError(err) {
+	if err := store.Get(ctx, preset[1].key, "", &example.Pod{}, false); !storage.IsNotFound(err) {
```

Comment:
This matches the existing behavior: 

https://github.com/kubernetes/kubernetes/blob/08afeb855ffa9da78b862121aa38b0da7936dbe8/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go#L743-L750